src/js/LineChart.jsx

```diff
@@ -0,0 +1,294 @@
+        </text>
+        <text
+          transform={`translate(${svgWidth}, ${svgHeight})`}
+          textAnchor="end"
+        >
+          {[this.props.data[coin][duration].length - 1].d}
+        </text>
+      </g>
+    );
+  }
+  // FIND CLOSEST POINT TO MOUSE
+  getCoords(e) {
+    const { svgWidth, data, yLabelSize, duration, coin } = this.props;
+    const svgLocation = document
+      .getElementsByClassName('linechart')[0]
+      .getBoundingClientRect();
+    const adjustment = (svgLocation.width - svgWidth) / 2; //takes padding into consideration
+    const relativeLoc = e.clientX - svgLocation.left - adjustment;
+
+    let svgData = [];
+    data[coin][duration].map((point, i) => {
+      svgData.push({
+        svgX: this.getSvgX(point.x),
+        svgY: this.getSvgY(point.y),
+        d: point.d,
+        p: point.p
+      });
+    });
+
+    let closestPoint = {};
+    for (let i = 0, c = 500; i < svgData.length; i++) {
+      if (Math.abs(svgData[i].svgX - this.props.hoverLoc) <= c) {
+        c = Math.abs(svgData[i].svgX - this.props.hoverLoc);
+        closestPoint = svgData[i];
+      }
+    }
+
+    if (relativeLoc - yLabelSize < 0) {
+      this.stopHover();
+    } else {
+      this.props.handleHover(relativeLoc, closestPoint);
+    }
+    this.props.onChartHover(relativeLoc, closestPoint);
+  }
+
+  // STOP HOVER
+  stopHover() {
+    this.props.handleHover(null, null);
+    // ({ hoverLoc: null, activePoint: null });
+    this.props.onChartHover(null, null);
+  }
+  // MAKE ACTIVE POINT
+  makeActivePoint() {
+    const { color, pointRadius } = this.props;
+    return (
+      <circle
+        className="linechart_point"
+        style={{ stroke: color }}
+        r={pointRadius}
+        cx={this.props.activePoint.svgX}
+        cy={this.props.activePoint.svgY}
+      />
+    );
+  }
+  // MAKE HOVER LINE
+  createLine() {
+    const { svgHeight, xLabelSize } = this.props;
+    return (
+      <line
+        className="hoverLine"
+        x1={this.props.hoverLoc}
+        y1={-8}
+        x2={this.props.hoverLoc}
+        y2={svgHeight - xLabelSize}
+      />
+    );
+  }
+
+  render() {
+    const { svgHeight, svgWidth } = this.props;
+
+    return (
+      <svg
+        width={svgWidth}
+        height={svgHeight}
+        viewBox={`0 0 ${svgWidth} ${svgHeight}`}
+        className="linechart"
+        onMouseLeave={() => this.stopHover()}
+        onMouseMove={e => this.getCoords(e)}
+      >
+        <g>
+          {this.makeAxis()}
+          {this.makePath()}
+          {this.makeArea()}
+          {this.makeLabels()}
+          {this.props.hoverLoc ? this.createLine() : null}
+          {this.props.hoverLoc ? this.makeActivePoint() : null}
+        </g>
+      </svg>
+    );
+  }
+}
+// DEFAULT PROPS
+LineChart.defaultProps = {
+  data: [],
+  color: '#fc4a1a',
+  pointRadius: 5,
+  svgHeight: 300,
+  svgWidth: 900,
+  xLabelSize: 20,
+  yLabelSize: 80
+};
+
+const mapStateToProps = state => ({
```

Comment:
This could be rewritten as:
```js
const mapStateToProps = ({duration, data, coin, hoverLoc, activePoint}) => ({
   duration,
   data,
   coin,
   hoverLoc,
   activePoint
 });
```